src/Core/Layers/InputLayer.hh

```diff
@@ -0,0 +1,23 @@
+#ifndef LAYERS_INPUT_LAYER_HH
+#define LAYERS_INPUT_LAYER_HH
+
+#include "Core/Events/Events.hh"
+#include "Layer.hh"
+
+namespace esp
+{
+  class InputLayer : public Layer
+  {
+   public:
+    InputLayer() = default;
+
+    void handle_event(Event& event, float dt) override;
+
+   private:
+    bool key_pressed_event_handler(KeyPressedEvent& event, float dt);
```

Comment:
You're checking only for key presed events. This will renault in camera stopping for a while after initial key press. To achieve a more smooth camera movement you also need to check against key repeat events.

Reply:
Not really imo. `glfwGetKey` handles both cases well (and in case of keyboard input it works much better than `glfwSetKeyCallback` - that's why I changed it). Perhaps you could run the code and test it out by yourself

Reply:
Right, my bad. Seems like camera movement is handled correctly.